### ISSUE TITLE
Forbid/remove tabs doesn't like .gitmodules

### ISSUE THREAD

**Issue by henryiii** [Lucas-C/pre-commit-hooks#14]
I believe `.gitmodules` should be excluded by default as well.

By the way, in the documentation, `sha:` is given, but that was replaced by `rev:` for pre-commit a long time ago (1.7+). It still works, but it would be nice to show the newer, recommended name. Ideally should be changed in the little script shown in the readme, too, I think.

**Comment by Lucas-C** [comment]
Thanks for the PR and for notifiying me about the doc!
This is fixed now, and released in `v1.1.8` :)

**Comment by henryiii** [comment]
Thanks! I tried to take advantage of this:

https://github.com/pybind/pybind11/pull/2311/commits/edd96218fe496c71061ae0eac1676aca799f626b

But even thought this seems like it should be exactly the same as what happens now, I get a format error about the gitmodules:

https://github.com/pybind/pybind11/runs/894400061

Do you have an idea why this would be different? Mayne the `.` needs to be escaped, but then why didn't it need to be before?

**Comment by Lucas-C** [comment]
Could you try running it with `--show-diff-on-failure --verbose` ?

**Comment by henryiii** [comment]
Doesn't really add much that seems useful. You can try it yourself with:

```bash
git clone git@github.com:pybind/pybind11 --branch ci/gha3
```

<details><summary>Click to expand output</summary>

```
pre-commit run --all-files --show-diff-on-failure --verbose 
[INFO] Initializing environment for https://github.com/Lucas-C/pre-commit-hooks.
[INFO] Installing environment for https://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Check for added large files..............................................Passed
- hook id: check-added-large-files
- duration: 0.18s
Check for case conflicts.................................................Passed
- hook id: check-case-conflict
- duration: 0.14s
Check for merge conflicts................................................Passed
- hook id: check-merge-conflict
- duration: 0.11s
Check for broken symlinks............................(no files to check)Skipped
- hook id: check-symlinks
Check Yaml...............................................................Passed
- hook id: check-yaml
- duration: 0.08s
Debug Statements (Python)................................................Passed
- hook id: debug-statements
- duration: 0.13s
Fix End of Files.........................................................Passed
- hook id: end-of-file-fixer
- duration: 0.12s
Mixed line ending........................................................Passed
- hook id: mixed-line-ending
- duration: 0.12s
Fix requirements.txt.....................................................Passed
- hook id: requirements-txt-fixer
- duration: 0.05s
Trim Trailing Whitespace.................................................Passed
- hook id: trailing-whitespace
- duration: 0.11s
Fix python encoding pragma...............................................Passed
- hook id: fix-encoding-pragma
- duration: 0.1s
Tabs remover.............................................................Failed
- hook id: remove-tabs
- duration: 0.1s
- exit code: 1
- files were modified by this hook

Substituting tabs in: .gitmodules by 4 whitespaces

Tabs have been successfully removed. Now aborting the commit.
You can check the changes made. Then simply "git add --update ." and re-commit

flake8...................................................................Passed
- hook id: flake8
- duration: 2.84s
Classic check-style......................................................Passed
- hook id: check-style
- duration: 0.12s
pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/.gitmodules b/.gitmodules
index d063a8e..8dcddb8 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "tools/clang"]
-	path = tools/clang
-	url = ../../wjakob/clang-cindex-python3
+    path = tools/clang
+    url = ../../wjakob/clang-cindex-python3
```

</details>

**Comment by Lucas-C** [comment]
This should be properly solved in `v1.1.9`.
The issue was that your PR fixed the legacy format file, `hooks.yaml`, instead of `.pre-commit-hooks.yaml`.

**Comment by henryiii** [comment]
Fantastic, thanks!
